SDWebImage/SDWebImageWebPCoder.m

```diff
@@ -347,6 +347,14 @@ - (nonnull CGColorSpaceRef)sd_colorSpaceWithDemuxer:(nonnull WebPDemuxer *)demux
             NSData *profileData = [NSData dataWithBytes:chunk_iter.chunk.bytes length:chunk_iter.chunk.size];
             colorSpaceRef = CGColorSpaceCreateWithICCProfile((__bridge CFDataRef)profileData);
             WebPDemuxReleaseChunkIterator(&chunk_iter);
+            if (colorSpaceRef) {
+                // `CGImageCreate` does not support colorSpace other than RGB (such as Monochrome), we must filter the colorSpace mode
+                CGColorSpaceModel model = CGColorSpaceGetModel(colorSpaceRef);
+                if (model != kCGColorSpaceModelRGB) {
```

Comment:
I don't think `CGImageCreate` don't support `Monochrome`, two reasons leads created failed, one is we passed wrong `bitsPerPixel`, `Monochrome` has only one component, you can try `CGImageRef imageRef = CGImageCreate(width, height, 8, 8, config.output.u.RGBA.stride, colorSpaceRef, bitmapInfo, provider, NULL, NO, renderingIntent);` , at least, image can show proportional. Other reason, it's also why it's show proportional is we already decode it to RGB data.

<img width="209" alt="image" src="https://user-images.githubusercontent.com/5061845/53466757-3e823b00-3a8e-11e9-93dc-4bc11def9ee6.png">



![](https://user-images.githubusercontent.com/6919743/53389219-69ef2200-39c9-11e9-9ee6-97d24b9be06a.png)

Reply:
......Maybe you misunderstand something.

We ask libwebp, to output a RGB/RGBA bitmap buffer. Even it's original encoded with Monochrome(grey style) which contains 1 component. The ouput bitmap buffer is always 3 components (transformed by libwebp from Monochrome to RGB).

If you supply 1 component to CGImageCreate, your dataProvider, actually is wrong. Because the bitmapBuffer is RGB(3 components), the colorSpace is Monochrome(1 component). It's also not correct

Actually, you can use ColorSync and vImage, to convert a ICC Profile from RGB to Monochrome, or even CMYK(YUV), but I don't think it's useful. We have never designed to be a professional image processing tools or decoding tools. If you build apps like Photoshop iOS and your customer is worry about the color sync, you should do that. This is just fix the rare cases during WebP image on the Internet.

Reply:
@dreampiggy ...Emm, maybe you don't get what I want to say 😂 , I only want to explain why `CGImageCreate` failed, there have two reasons. So the comments in this `PR` is wrong, I made a `PR` #2629  to update.